src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -1856,11 +1844,19 @@ void SolverInterfaceImpl::performDataActions(
     double                                  time)
 {
   PRECICE_TRACE();
+  // for actions we need to load and write back data from/to time steps storage.
+  // Actions would need similar treatment like mapping.
+  if (_couplingScheme->hasDataBeenReceived()) {
+    _couplingScheme->loadReceiveDataFromStorage(time::Storage::WINDOW_END);
+  }
   for (action::PtrAction &action : _accessor->actions()) {
     if (timings.find(action->getTiming()) != timings.end()) {
       action->performAction(time);
     }
   }
+  if (_couplingScheme->hasDataBeenReceived()) {
+    _couplingScheme->storeReceiveData(time::Storage::WINDOW_END);
+  }
```

Comment:
In #1523 the current implementation of actions turned out to become more and more problematic. So I removed them via [91d76c6](https://github.com/precice/precice/pull/1523/commits/91d76c678f246670d3fedf0c1034bfb46d84ae62). This here is basically already the first step into the problematic direction. I guess we have to try to find a better solution here.